src/frontend/src/handler/util.rs

```diff
@@ -41,9 +41,9 @@ fn pg_value_format(d: ScalarRefImpl, format: bool) -> Bytes {
             _ => d.to_string().into(),
         }
     } else {
-        let mut serializer = Serializer::new();
-        d.binary_serialize(&mut serializer).unwrap();
-        serializer.get_ouput()
+        let mut encoder = BinaryEncoder::new();
+        d.binary_serialize(&mut encoder);
```

Comment:
What about make encoder accept a ScalarRefImpl instead versa?

Otherwise common will include pgwire, instead of pgwire include common 😢 .

Or we move the encoder into common instead of pgwire

Reply:
Maybe previously we misunderstand some points...

Reply:
Yes